vhd_format/f.ml

```diff
@@ -1465,6 +1465,13 @@ module From_input = functor (I: S.INPUT) -> struct
         sector 0 (fun () -> block (M.remove s blocks) andthen) in
     block phys_to_virt (fun () ->
     let buffer = alloc Footer.sizeof in
+    ( match size_opt with
+    | None -> return ()
+    | Some s ->
+        let (&&&)  = Int64.logand in
+        let footer_offset = Int64.(sub s 1L &&& lognot 0b1_1111_1111L) in
+        (* offset is last 512-byte-aligned block in the file *)
+        skip_to fd footer_offset) >>= fun () ->
```

Comment:
I was worried that we might not be able to seek back if this is an actual stream, but it looks like reading the footer is the last thing we do, so this is probably right